src/Mobs/Path.cpp

```diff
@@ -139,11 +149,25 @@ bool cPath::StepOnce()
 	// Path found.
 	if (CurrentCell->m_Location == m_Destination)
 	{
-		BuildPath();
+		BuildPath(m_Destination);
 		FinishCalculation(ePathFinderStatus::PATH_FOUND);
 		return true;
 	}
 
+	if (m_IsWideDestination)
+	{
+		Vector3i Delta = CurrentCell->m_Location - m_Destination;
+		int DeltaX = abs(Delta.x);
+		int DeltaZ = abs(Delta.z);
+		if ((Delta.y == 0) && (((DeltaX == 0) && (DeltaZ == 1)) || ((DeltaX == 1) && (DeltaZ == 0))))
```

Comment:
This is a nasty condition. It might be worth breaking out the `(DeltaX == 0) && (DeltaZ == 1)` clauses on to separate lines, and use indentation to make it clear the `(Delta.y == 0)` clause applies to both.